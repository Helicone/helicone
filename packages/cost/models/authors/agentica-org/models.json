{
  "deepcoder-14b-preview:free": {
    "id": "deepcoder-14b-preview:free",
    "name": "Agentica: Deepcoder 14B Preview (free)",
    "author": "agentica-org",
    "description": "DeepCoder-14B-Preview is a 14B parameter code generation model fine-tuned from DeepSeek-R1-Distill-Qwen-14B using reinforcement learning with GRPO+ and iterative context lengthening. It is optimized for long-context program synthesis and achieves strong performance across coding benchmarks, including 60.6% on LiveCodeBench v5, competitive with models like o3-Mini",
    "contextLength": 96000,
    "maxOutputTokens": null,
    "created": "2025-04-13T14:43:15.000Z",
    "modality": "text->text",
    "tokenizer": "Other"
  },
  "deepcoder-14b-preview": {
    "id": "deepcoder-14b-preview",
    "name": "Agentica: Deepcoder 14B Preview",
    "author": "agentica-org",
    "description": "DeepCoder-14B-Preview is a 14B parameter code generation model fine-tuned from DeepSeek-R1-Distill-Qwen-14B using reinforcement learning with GRPO+ and iterative context lengthening. It is optimized for long-context program synthesis and achieves strong performance across coding benchmarks, including 60.6% on LiveCodeBench v5, competitive with models like o3-Mini",
    "contextLength": 96000,
    "maxOutputTokens": null,
    "created": "2025-04-13T14:43:15.000Z",
    "modality": "text->text",
    "tokenizer": "Other"
  }
}