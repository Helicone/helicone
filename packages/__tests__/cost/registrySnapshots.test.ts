import { describe, it, expect, beforeAll } from "@jest/globals";
import { sync as globSync } from "glob";
import * as path from "path";
import { buildIndexes } from "../../cost/models/build-indexes";
import { getUsageProcessor } from "@/cost/usage/getUsageProcessor";

// Dynamically discover and import all endpoint files
async function loadAllEndpoints() {
  const files = globSync("**/endpoints.ts", {
    cwd: path.join(__dirname, "../../cost/models/authors"),
  });

  const endpoints: Record<string, any> = {};
  const archivedEndpoints: Record<string, any> = {};

  for (const file of files) {
    try {
      const [author, model] = file.split("/");
      const key = `${author}/${model}`;
      const module = await import(
        path.join(__dirname, "../../cost/models/authors", file)
      );
      if (module.endpoints) {
        endpoints[key] = module.endpoints;
      }
      if (module.archivedEndpoints) {
        archivedEndpoints[key] = module.archivedEndpoints;
      }
    } catch (error) {
      console.warn(`Failed to import ${file}:`, error);
    }
  }

  return { endpoints, archivedEndpoints };
}

describe("Registry Snapshots", () => {
  let allEndpoints: Record<string, any>;
  let allArchivedEndpoints: Record<string, any>;

  beforeAll(async () => {
    const loaded = await loadAllEndpoints();
    allEndpoints = loaded.endpoints;
    allArchivedEndpoints = loaded.archivedEndpoints;
  });

  it("pricing snapshot", () => {
    const snapshot: Record<string, any> = {};

    Object.entries(allEndpoints).forEach(([model, endpoints]) => {
      snapshot[model] = {};
      Object.entries(endpoints).forEach(([key, config]: [string, any]) => {
        const provider = key.split(":").pop()!;
        snapshot[model][provider] = config.pricing;
      });
    });

    expect(snapshot).toMatchSnapshot();
  });

  it("model coverage snapshot", () => {
    const coverage: Record<string, string[]> = {};

    Object.entries(allEndpoints).forEach(([model, endpoints]) => {
      coverage[model] = Object.keys(endpoints)
        .map((key) => key.split(":").pop()!)
        .filter(Boolean)
        .sort();
    });

    expect(coverage).toMatchSnapshot();
  });

  it("endpoint configurations snapshot", () => {
    const configs: Record<string, any> = {};

    Object.entries(allEndpoints).forEach(([model, endpoints]) => {
      configs[model] = {};
      Object.entries(endpoints).forEach(([key, config]: [string, any]) => {
        configs[model][key] = {
          provider: config.provider,
          modelId: config.providerModelId,
          context: config.contextLength,
          maxTokens: config.maxCompletionTokens,
          ptbEnabled: config.ptbEnabled,
          crossRegion: config.crossRegion || false,
          regions: Object.keys(config.endpointConfigs).sort(),
          parameters: config.supportedParameters.sort(),
        };
      });
    });

    expect(configs).toMatchSnapshot();
  });

  it("verify registry state", () => {
    // Flatten for buildIndexes
    const flat: Record<string, any> = {};
    Object.values(allEndpoints).forEach((endpoints) => {
      Object.assign(flat, endpoints);
    });

    // Flatten archived endpoints
    const flatArchived: Record<string, any> = {};
    Object.values(allArchivedEndpoints).forEach((archivedEndpoints) => {
      Object.assign(flatArchived, archivedEndpoints);
    });

    const indexes = buildIndexes(flat, flatArchived);

    // Verify all the important maps are built
    const snapshot = {
      // Total counts
      totalEndpoints: indexes.endpointIdToEndpoint.size,
      totalModelProviderConfigs: indexes.endpointConfigIdToEndpointConfig.size,
      totalProviders: indexes.providerToModels.size,
      totalModelsWithPtb: indexes.modelToPtbEndpoints.size,

      // Provider breakdown - which providers have how many models
      providerBreakdown: Array.from(indexes.providerToModels.entries())
        .map(([provider, models]) => ({ provider, modelCount: models.size }))
        .sort((a, b) => a.provider.localeCompare(b.provider)),

      // Model to providers mapping - which models are available from which providers
      modelToProviders: Array.from(indexes.modelToProviders.entries())
        .map(([model, providers]) => ({
          model,
          providers: Array.from(providers).sort(),
        }))
        .sort((a, b) => a.model.localeCompare(b.model)),

      // PTB-enabled models
      ptbEnabledModels: Array.from(indexes.modelToPtbEndpoints.keys()).sort(),

      // Sample endpoint IDs to ensure format is correct
      sampleEndpointIds: Array.from(indexes.endpointIdToEndpoint.keys())
        .sort()
        .slice(0, 5),

      // Archived endpoints index
      totalArchivedConfigs: indexes.modelToArchivedEndpointConfigs.size,

      // Sample archived endpoint keys to ensure format is correct
      sampleArchivedKeys: Array.from(indexes.modelToArchivedEndpointConfigs.keys())
        .sort()
        .slice(0, 5),
    };

    // Verify all PTB endpoints have usage processors
    const modelKeys = Array.from(indexes.modelToPtbEndpoints.keys());
    modelKeys.forEach((key) => {
      const endpoints = indexes.modelToPtbEndpoints.get(key);
      if (!endpoints) {
        throw new Error(`No endpoints found for model: ${key}`);
      }
      endpoints.forEach((endpoint) => {
        const usageProcessor = getUsageProcessor(endpoint.provider);
        if (!usageProcessor) {
          throw new Error(`Usage processor not found for provider: "${endpoint.provider}" in model: "${key}"`);
        }
      });
    });

    expect(snapshot).toMatchSnapshot();
  });

  it("archived endpoints snapshot", () => {
    const archivedSnapshot: Record<string, any> = {};

    Object.entries(allArchivedEndpoints).forEach(([model, archivedEndpoints]) => {
      if (Object.keys(archivedEndpoints).length > 0) {
        archivedSnapshot[model] = {};
        Object.entries(archivedEndpoints).forEach(([key, config]: [string, any]) => {
          archivedSnapshot[model][key] = {
            provider: config.provider,
            version: config.version,
            modelId: config.providerModelId,
            pricing: config.pricing,
          };
        });
      }
    });

    expect(archivedSnapshot).toMatchSnapshot();
  });
});
