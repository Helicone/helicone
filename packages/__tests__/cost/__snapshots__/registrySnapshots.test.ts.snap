// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Snapshots endpoint configurations snapshot 1`] = `
{
  "anthropic/claude-3.5-haiku": {
    "claude-3.5-haiku:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 8192,
      "modelId": "claude-3-5-haiku-20241022",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-3.5-haiku:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 8192,
      "modelId": "anthropic.claude-3-5-haiku-20241022-v1:0",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-3.5-haiku:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 8192,
      "modelId": "claude-3-5-haiku@20241022",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "vertex",
      "ptbEnabled": false,
      "regions": [
        "global",
      ],
    },
  },
  "anthropic/claude-3.5-sonnet-v2": {
    "claude-3.5-sonnet-v2:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 8192,
      "modelId": "claude-3-5-sonnet-20241022",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-3.5-sonnet-v2:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 8192,
      "modelId": "anthropic.claude-3-5-sonnet-20241022-v2:0",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-3.5-sonnet-v2:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 8192,
      "modelId": "claude-3-5-sonnet-v2@20241022",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "vertex",
      "ptbEnabled": true,
      "regions": [
        "global",
      ],
    },
  },
  "anthropic/claude-3.7-sonnet": {
    "claude-3.7-sonnet:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 64000,
      "modelId": "claude-3-7-sonnet-20250219",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-3.7-sonnet:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 64000,
      "modelId": "anthropic.claude-3-7-sonnet-20250219-v1:0",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-3.7-sonnet:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 64000,
      "modelId": "claude-3-7-sonnet@20250219",
      "parameters": [
        "max_tokens",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "vertex",
      "ptbEnabled": true,
      "regions": [
        "global",
      ],
    },
  },
  "anthropic/claude-opus-4": {
    "claude-opus-4:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 32000,
      "modelId": "claude-opus-4-20250514",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-opus-4:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 32000,
      "modelId": "anthropic.claude-opus-4-20250514-v1:0",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-opus-4:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 32000,
      "modelId": "claude-opus-4@20250514",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "vertex",
      "ptbEnabled": true,
      "regions": [
        "global",
      ],
    },
  },
  "anthropic/claude-opus-4-1": {
    "claude-opus-4-1:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 32000,
      "modelId": "claude-opus-4-1-20250805",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-opus-4-1:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 32000,
      "modelId": "anthropic.claude-opus-4-1-20250805-v1:0",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-opus-4-1:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 32000,
      "modelId": "claude-opus-4-1@20250805",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "vertex",
      "ptbEnabled": true,
      "regions": [
        "global",
      ],
    },
  },
  "anthropic/claude-sonnet-4": {
    "claude-sonnet-4:anthropic": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 64000,
      "modelId": "claude-sonnet-4-20250514",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "anthropic",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "claude-sonnet-4:bedrock": {
      "context": 200000,
      "crossRegion": true,
      "maxTokens": 64000,
      "modelId": "anthropic.claude-sonnet-4-20250514-v1:0",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_k",
        "top_p",
      ],
      "provider": "bedrock",
      "ptbEnabled": true,
      "regions": [
        "us-east-1",
      ],
    },
    "claude-sonnet-4:vertex": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 64000,
      "modelId": "claude-sonnet-4@20250514",
      "parameters": [
        "include_reasoning",
        "max_tokens",
        "reasoning",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
      ],
      "provider": "vertex",
      "ptbEnabled": true,
      "regions": [
        "global",
      ],
    },
  },
  "openai/gpt-4": {
    "gpt-4:openai": {
      "context": 8191,
      "crossRegion": false,
      "maxTokens": 4096,
      "modelId": "gpt-4",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
  "openai/gpt-4.1": {
    "gpt-4.1-mini:openai": {
      "context": 1047576,
      "crossRegion": false,
      "maxTokens": 32768,
      "modelId": "gpt-4.1-mini",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4.1-nano:openai": {
      "context": 1047576,
      "crossRegion": false,
      "maxTokens": 32768,
      "modelId": "gpt-4.1-nano",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4.1:openai": {
      "context": 1047576,
      "crossRegion": false,
      "maxTokens": 32768,
      "modelId": "gpt-4.1",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
  "openai/gpt-4o": {
    "chatgpt-4o-latest:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 16384,
      "modelId": "chatgpt-4o-latest",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4o-mini-search-preview:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 16384,
      "modelId": "gpt-4o-mini-search-preview",
      "parameters": [
        "max_tokens",
        "response_format",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4o-mini:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 16384,
      "modelId": "gpt-4o-mini",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4o-search-preview:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 16384,
      "modelId": "gpt-4o-search-preview",
      "parameters": [
        "max_tokens",
        "response_format",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4o:extended:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 64000,
      "modelId": "gpt-4o-2024-08-06",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "gpt-4o:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 16384,
      "modelId": "gpt-4o",
      "parameters": [
        "frequency_penalty",
        "max_tokens",
        "presence_penalty",
        "response_format",
        "seed",
        "stop",
        "temperature",
        "tool_choice",
        "tools",
        "top_p",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
  "openai/o1": {
    "o1-mini:openai": {
      "context": 128000,
      "crossRegion": false,
      "maxTokens": 65536,
      "modelId": "o1-mini",
      "parameters": [
        "max_tokens",
        "seed",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o1-pro:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o1-pro",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o1:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o1",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
  "openai/o3": {
    "o3-mini-high:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o3-mini-high",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o3-mini:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o3-mini",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o3-pro:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o3-pro-2025-06-10",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o3:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o3-2025-04-16",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
  "openai/o4": {
    "o4-mini-high:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o4-mini-high-2025-04-16",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
    "o4-mini:openai": {
      "context": 200000,
      "crossRegion": false,
      "maxTokens": 100000,
      "modelId": "o4-mini",
      "parameters": [
        "max_tokens",
        "response_format",
        "seed",
        "tool_choice",
        "tools",
      ],
      "provider": "openai",
      "ptbEnabled": true,
      "regions": [
        "*",
      ],
    },
  },
}
`;

exports[`Registry Snapshots model coverage snapshot 1`] = `
{
  "anthropic/claude-3.5-haiku": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "anthropic/claude-3.5-sonnet-v2": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "anthropic/claude-3.7-sonnet": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "anthropic/claude-opus-4": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "anthropic/claude-opus-4-1": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "anthropic/claude-sonnet-4": [
    "anthropic",
    "bedrock",
    "vertex",
  ],
  "openai/gpt-4": [
    "openai",
  ],
  "openai/gpt-4.1": [
    "openai",
    "openai",
    "openai",
  ],
  "openai/gpt-4o": [
    "openai",
    "openai",
    "openai",
    "openai",
    "openai",
    "openai",
  ],
  "openai/o1": [
    "openai",
    "openai",
    "openai",
  ],
  "openai/o3": [
    "openai",
    "openai",
    "openai",
    "openai",
  ],
  "openai/o4": [
    "openai",
    "openai",
  ],
}
`;

exports[`Registry Snapshots pricing snapshot 1`] = `
{
  "anthropic/claude-3.5-haiku": {
    "anthropic": {
      "cacheRead": 8e-8,
      "cacheWrite": {
        "1h": 0.0000016,
        "5m": 0.000001,
        "default": 0.000001,
      },
      "completion": 0.000004,
      "prompt": 8e-7,
    },
    "bedrock": {
      "cacheRead": 8e-8,
      "cacheWrite": 0.000001,
      "completion": 0.000004,
      "prompt": 8e-7,
    },
    "vertex": {
      "cacheRead": 8e-8,
      "cacheWrite": 0.000001,
      "completion": 0.000004,
      "prompt": 8e-7,
    },
  },
  "anthropic/claude-3.5-sonnet-v2": {
    "anthropic": {
      "cacheRead": 3e-7,
      "cacheWrite": {
        "1h": 0.000006,
        "5m": 0.00000375,
        "default": 0.00000375,
      },
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "bedrock": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "vertex": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
  },
  "anthropic/claude-3.7-sonnet": {
    "anthropic": {
      "cacheRead": 3e-7,
      "cacheWrite": {
        "1h": 0.000006,
        "5m": 0.00000375,
        "default": 0.00000375,
      },
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "bedrock": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "vertex": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
  },
  "anthropic/claude-opus-4": {
    "anthropic": {
      "cacheRead": 0.0000015,
      "cacheWrite": {
        "1h": 0.00003,
        "5m": 0.00001875,
        "default": 0.00001875,
      },
      "completion": 0.000075,
      "prompt": 0.000015,
    },
    "bedrock": {
      "cacheRead": 0.0000015,
      "cacheWrite": 0.00001875,
      "completion": 0.000075,
      "prompt": 0.000015,
    },
    "vertex": {
      "cacheRead": 0.0000015,
      "cacheWrite": 0.00001875,
      "completion": 0.000075,
      "prompt": 0.000015,
    },
  },
  "anthropic/claude-opus-4-1": {
    "anthropic": {
      "cacheRead": 0.0000015,
      "cacheWrite": {
        "1h": 0.00003,
        "5m": 0.00001875,
        "default": 0.00001875,
      },
      "completion": 0.000075,
      "prompt": 0.000015,
    },
    "bedrock": {
      "cacheRead": 0.0000015,
      "cacheWrite": 0.00001875,
      "completion": 0.000075,
      "prompt": 0.000015,
    },
    "vertex": {
      "cacheRead": 0.0000015,
      "cacheWrite": 0.00001875,
      "completion": 0.000075,
      "prompt": 0.000015,
    },
  },
  "anthropic/claude-sonnet-4": {
    "anthropic": {
      "cacheRead": 3e-7,
      "cacheWrite": {
        "1h": 0.000006,
        "5m": 0.00000375,
        "default": 0.00000375,
      },
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "bedrock": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
    "vertex": {
      "cacheRead": 3e-7,
      "cacheWrite": 0.00000375,
      "completion": 0.000015,
      "prompt": 0.000003,
    },
  },
  "openai/gpt-4": {
    "openai": {
      "completion": 0.00006,
      "prompt": 0.00003,
    },
  },
  "openai/gpt-4.1": {
    "openai": {
      "cacheRead": 5e-8,
      "completion": 4e-7,
      "prompt": 1e-7,
    },
  },
  "openai/gpt-4o": {
    "openai": {
      "completion": 0.00001,
      "prompt": 0.0000025,
    },
  },
  "openai/o1": {
    "openai": {
      "cacheRead": 5.5e-7,
      "completion": 0.0000044,
      "prompt": 0.0000011,
    },
  },
  "openai/o3": {
    "openai": {
      "cacheRead": 5.5e-7,
      "completion": 0.0000044,
      "prompt": 0.0000011,
    },
  },
  "openai/o4": {
    "openai": {
      "cacheRead": 2.75e-7,
      "completion": 0.0000044,
      "prompt": 0.0000011,
    },
  },
}
`;

exports[`Registry Snapshots registry builds correctly 1`] = `
{
  "modelToProviders": [
    {
      "model": "chatgpt-4o-latest",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "claude-3.5-haiku",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "claude-3.5-sonnet-v2",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "claude-3.7-sonnet",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "claude-opus-4",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "claude-opus-4-1",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "claude-sonnet-4",
      "providers": [
        "anthropic",
        "bedrock",
        "vertex",
      ],
    },
    {
      "model": "gpt-4",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4.1",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4.1-mini",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4.1-nano",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4o",
      "providers": [
        "extended",
        "openai",
      ],
    },
    {
      "model": "gpt-4o-mini",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4o-mini-search-preview",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "gpt-4o-search-preview",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o1",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o1-mini",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o1-pro",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o3",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o3-mini",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o3-mini-high",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o3-pro",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o4-mini",
      "providers": [
        "openai",
      ],
    },
    {
      "model": "o4-mini-high",
      "providers": [
        "openai",
      ],
    },
  ],
  "providerBreakdown": [
    {
      "modelCount": 6,
      "provider": "anthropic",
    },
    {
      "modelCount": 6,
      "provider": "bedrock",
    },
    {
      "modelCount": 1,
      "provider": "extended",
    },
    {
      "modelCount": 18,
      "provider": "openai",
    },
    {
      "modelCount": 6,
      "provider": "vertex",
    },
  ],
  "ptbEnabledModels": [
    "chatgpt-4o-latest",
    "claude-3.5-haiku",
    "claude-3.5-sonnet-v2",
    "claude-3.7-sonnet",
    "claude-opus-4",
    "claude-opus-4-1",
    "claude-sonnet-4",
    "gpt-4",
    "gpt-4.1",
    "gpt-4.1-mini",
    "gpt-4.1-nano",
    "gpt-4o",
    "gpt-4o-mini",
    "gpt-4o-mini-search-preview",
    "gpt-4o-search-preview",
    "o1",
    "o1-mini",
    "o1-pro",
    "o3",
    "o3-mini",
    "o3-mini-high",
    "o3-pro",
    "o4-mini",
    "o4-mini-high",
  ],
  "sampleEndpointIds": [
    "claude-3.5-haiku:anthropic:*",
    "claude-3.5-haiku:bedrock:us-east-1",
    "claude-3.5-haiku:vertex:global",
    "claude-3.5-sonnet-v2:anthropic:*",
    "claude-3.5-sonnet-v2:vertex:global",
  ],
  "totalEndpoints": 37,
  "totalModelProviderConfigs": 37,
  "totalModelsWithPtb": 24,
  "totalProviders": 5,
}
`;
